org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/gradle/GradleProgressAttributes.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2018 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.buildship.core.gradle;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+
+import org.gradle.tooling.CancellationToken;
+import org.gradle.tooling.CancellationTokenSource;
+import org.gradle.tooling.ProgressListener;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.StandardSystemProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+
+import org.eclipse.buildship.core.CorePlugin;
+import org.eclipse.buildship.core.console.ProcessDescription;
+import org.eclipse.buildship.core.console.ProcessStreams;
+import org.eclipse.buildship.core.console.ProcessStreamsProvider;
+import org.eclipse.buildship.core.util.progress.CancellationForwardingListener;
+import org.eclipse.buildship.core.util.progress.DelegatingProgressListener;
+
+/**
+ * Holds attributes that are commonly used to handle progress in each Gradle invocation.
+ *
+ * @author Donat Csikos
+ */
+public final class GradleProgressAttributes {
```

Comment:
Just a thought here: Do we even need this parameter class or could we extend the `ConfigurableLauncher` interface from the TAPI instead by adding some Buildship-specific convenience methods?

Reply:
I'm not sure I understand the proposal. Do you mean we should reimplement the `ConfigurableLauncher` interface so that can apply the attributes on the build/test launchers created by `GradleBuild`? In a way, `GradleProgressAttributes` already does this.

Reply:
I was thinking that the methods on `GradleBuild` could just return an `ExtendedConfigurableLauncher extends ConfigurableLauncher` (forgive the dumb name). Then users can modify everything they can do on the TAPI, but also get some convenience methods on top.